c/tskit/tables.c

```diff
@@ -1343,6 +1343,44 @@ tsk_edge_table_load(tsk_edge_table_t *self, kastore_t *store)
     return read_table_cols(store, read_cols, sizeof(read_cols) / sizeof(*read_cols));
 }
 
+int
+tsk_edge_table_squash(tsk_edge_table_t *self)
+{
+    int k;
+    int ret = 0;
+    tsk_edge_t *edges = NULL;
+    edges = malloc(self->num_rows * sizeof(tsk_edge_t));
+    if (edges == NULL){
+        ret = TSK_ERR_NO_MEMORY;
+        goto out;
+    }
+    tsk_size_t num_output_edges;
```

Comment:
This isn't mandated by C99, but older versions insisted on variable declarations coming before code. We've stuck to this in tskit as a generally good idea. So, the top of the function should look like:
```c
  int k;
  int ret = 0;
  tsk_edge_t *edges = NULL;
  tsk_size_t num_output_edges; 

  edges = malloc(self->num_rows * sizeof(tsk_edge_t));
```
By convention, we keep a space between the variable declaration bit at the start of the function and the actual code.